core/src/main/java/org/fao/geonet/kernel/datamanager/draft/DraftMetadataUtils.java

```diff
@@ -655,6 +659,23 @@ public void cancelEditingSession(ServiceContext context, String id) throws Excep
 
                 // Unset METADATA_EDITING_CREATED_DRAFT flag
                 context.getUserSession().removeProperty(Geonet.Session.METADATA_EDITING_CREATED_DRAFT);
+
+                // Get the resource store
+                Store store = context.getBean("resourceStore", Store.class);
+
+                // Register synchronization to delete resources after commit
+                TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronization() {
+                    @Override
+                    public void afterCommit() {
+                        try {
+                            // Delete resources from the store
+                            store.delResources(context, intId);
+                        } catch (Exception e) {
+                            Log.error(Geonet.DATA_MANAGER, "Couldn't delete resources for draft " + id, e);
+                        }
+                    }
+                });
+
             } catch (Exception e) {
                 Log.error(Geonet.DATA_MANAGER, "Couldn't cleanup draft " + id, e);
```

Comment:
Should the exception be re-thrown so the transaction is rolled back?

If for example the ` metadataRatingByIpRepository.deleteAllById_MetadataId(intId);` call fails the operations will still be deleted as the exception is caught so the transaction doesn't rollback.

Reply:
I think it is better not to re-throw the error. 

Otherwise if we re-throw an error after 50% of the resources were deleted then it could rollback the metadata delete and then the user would be working with a partial working copy which would be bad.